### PR TITLE
feat(accordion): Add Accordion component

### PR DESCRIPTION
**Pull request by diogomateus**
### What this PR does
This PR adds the `Accordion` component and does minimal changes to `Card` component for improved rewriting of styles.

### How to test?
Run storybook book and visit [Accordion story page](http://localhost:9009/?path=/docs/jsx-accordion--accordion-story).

**Preview:**
https://github.com/getPopsure/dirty-swan/assets/4015038/7d5a210f-e6f4-41f0-93eb-a95f91b362bb

### Checklist:

- [ ] I reviewed my own code
- [ ] The changes align with the designs I received  
  Or give a reason why this does not apply:
- [ ] I have attached screenshot(s), video(s) or gif(s) showing that the solution is working as expected  
  Or give a reason why this does not apply:
- [ ] I have updated the task(s) status on Linear
- [ ] All new media is optimized (images, gifs, videos)

### Browser support

My code works in the following browsers:

- [ ] Firefox
- [ ] Chrome
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Comment by diogomateus** [comment]
Missing design feedback on hover/focus states but ready for review apart from that

File: src/lib/components/accordion/index.tsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+              height={isOpen ? 'auto' : 0}
+            >
+              <div 
+                className={classnamesUtil(
+                  classNames?.answer,
+                  'tc-grey-600 p16 pl0', 
+                  { 'pl16': !isDefaultVariant },
+                )}
+              >
+                {answer}
```
Comment (diogomateus) [inline src/lib/components/accordion/index.tsx:137]: There is no Markdown integration, that needs to be done in other repositories as it differs between website/app.

File: src/lib/components/accordion/style.module.scss [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    border-bottom: 0;
+  }
+}
+
+.buttonWrapper {
+  border-radius: 0!important;
+
+  &:hover {
+    background: $ds-primary-50;
+    outline: 0 solid transparent!important;
```
Comment (diogomateus) [inline src/lib/components/accordion/style.module.scss:24]: Sadly we do need these `!important` here to override the current styles.

File: src/lib/components/accordion/index.tsx (hunk 1 above)
Comment (kimkwanka) [reply src/lib/components/accordion/index.tsx:137]: Can you elaborate?

Comment (diogomateus) [reply src/lib/components/accordion/index.tsx:137]: Sure! On both [app](https://github.com/getPopsure/app/blob/main/src/components/markdown/index.tsx) and [website](https://github.com/getPopsure/website/blob/main/components/markdown/index.tsx) we have different custom markdown integrations so we'll need to configure components to have their own markdown on answers.

I'm talking about this part that wasn't implemented on design system in this PR:
https://github.com/getPopsure/app/blob/main/src/components/faq/index.tsx#L77

**Review by a-maffei — APPROVED** [review]

**Comment by a-maffei** [comment]
> Missing design feedback on hover/focus states but ready for review apart from that

I was wondering about the focus state of the first item (whether only the bottom margin should have a border, unlike the other items), but it seems you'll tackle it separately. Other than that, it looks good! 💅

image 10

**Pull request merged**